libraries/stdlib/jvm/runtime/kotlin/jvm/internal/SourceDebugExtension.kt

```diff
@@ -0,0 +1,18 @@
+/*
+ * Copyright 2010-2022 JetBrains s.r.o. and Kotlin Programming Language contributors.
+ * Use of this source code is governed by the Apache 2.0 license that can be found in the license/LICENSE.txt file.
+ */
+
+package kotlin.jvm.internal
+
+/**
+ * Provides a copy of the JVM attribute SourceDebugExtension on the class file.
+ * This annotation exists if and only if there is a SourceDebugExtension attribute on the class.
+ * To obtain the stored source mapping information, concatenate the strings in [value].
+ * This annotation is needed for tools which inspect the Kotlin bytecode via JVMTI,
+ * which does not always provide access to the SourceDebugExtension attribute.
+ */
+@Target(AnnotationTarget.CLASS)
+@Retention(AnnotationRetention.BINARY)
+@SinceKotlin("1.8")
+annotation class SourceDebugExtension(val value: Array<String>)
```

Comment:
This is array of strings, because strings in JVM constant pool are `u2`, while `SourceDebugExtension` attribute value is `u4`.

Reply:
Will it be possible to locate SMAP in the constant pool as the last string starting with "SMAP\n" when the value is split into parts? Or, searching for annotation in the full class bytecode is the only way in such case?

Reply:
Hm, good point. I guess you can start from the string that starts with `"SMAP\n"` and concatenate it up to the string that ends with `*E`. :) These parts are guaranteed to be written one after the other in the constant pool. The only exception is if one part of this string is used somewhere else in the class file, but that seems impossible to achieve unless you're trying really hard to specifically construct such case.